x/application/keeper/application_index.go

```diff
@@ -0,0 +1,195 @@
+package keeper
+
+import (
+	"context"
+
+	"github.com/pokt-network/poktroll/x/application/types"
+)
+
+const ALL_UNDELEGATIONS = ""
+
+// indexApplicationUnstakingHeight maintains an index of applications that are
+// currently in the unbonding period.
+//
+// This function either adds or removes an application from the unstaking height index
+// depending on whether the application is currently unbonding:
+// - If the application is unbonding (UnstakeSessionEndHeight > 0), it's added to the index
+// - If the application is not unbonding, it's removed from the index
+//
+// DEV_NOTE: Since an unstaking application cannot perform successive unbonding
+// actions until it re-stakes or completes the unbonding period, we can safely
+// use the application's address as the key in the index.
+//
+// TODO_IMPROVE: Consider having an unbondingHeight/address key for even more efficient lookups.
+// This involves processing the unbonding height here in addition to the unbonding EndBlocker.
+//
+// This index enables the EndBlocker to efficiently find applications that are unbonding
+// without iterating over and unmarshaling all applications in the store.
+func (k Keeper) indexApplicationUnstaking(ctx context.Context, app types.Application) {
+	appUnstakingHeightStore := k.getApplicationUnstakingStore(ctx)
```

Comment:
Does this still have anything to do with height? It seems like it's using the application address now.

Reply:
That was from an old variant. Leveraging heights would be interesting but involves the introduction of many more indexes. We might introduce it if we want to trade more storage for speed.